src/Neo.CLI/CLI/MainService.Tools.cs

```diff
@@ -417,62 +418,81 @@ private static string Base64Fixed(string str)
+                var sb = new StringBuilder();
+                var line = 0;
+
+                foreach (var instruct in new VMInstruction(bytes))
+                {
+                    if (instruct.OperandSize == 0)
+                        sb.AppendFormat("L{0:D04}:{1:X04} {2}{3}", line, instruct.Position, instruct.OpCode, Environment.NewLine);
+                    else
+                        sb.AppendFormat("L{0:D04}:{1:X04} {2,-10}{3}{4}", line, instruct.Position, instruct.OpCode, DecodeOperand(instruct), Environment.NewLine);
+                    line++;
+                }
+
+                return sb.ToString();
             }
-            catch (Exception)
+            catch
             {
                 return null;
             }
-            return ScriptsToOpCode(script);
         }
 
-        private string ScriptsToOpCode(Script script)
+        private string DecodeOperand(VMInstruction instruction)
         {
-            //Initialize all InteropService
-            var dic = new Dictionary<uint, string>();
-            ApplicationEngine.Services.ToList().ForEach(p => dic.Add(p.Value.Hash, p.Value.Name));
-
-            //Analyzing Scripts
-            var ip = 0;
-            Instruction instruction;
-            var result = new List<string>();
-            while (ip < script.Length && (instruction = script.GetInstruction(ip)) != null)
-            {
-                ip += instruction.Size;
-
-                var op = instruction.OpCode;
-
-                if (op.ToString().StartsWith("PUSHINT"))
-                {
-                    var operand = instruction.Operand.ToArray();
-                    result.Add($"{op} {new BigInteger(operand)}");
-                }
-                else if (op == OpCode.SYSCALL)
-                {
-                    var operand = instruction.Operand.ToArray();
-                    result.Add($"{op} {dic[BitConverter.ToUInt32(operand)]}");
-                }
-                else
-                {
-                    if (!instruction.Operand.IsEmpty && instruction.Operand.Length > 0)
-                    {
-                        var operand = instruction.Operand.ToArray();
-                        var ascii = Encoding.Default.GetString(operand);
-                        ascii = ascii.Any(p => p < '0' || p > 'z') ? operand.ToHexString() : ascii;
-
-                        result.Add($"{op} {(operand.Length == 20 ? new UInt160(operand).ToString() : ascii)}");
-                    }
-                    else
-                    {
-                        result.Add($"{op}");
-                    }
-                }
-            }
-            return Environment.NewLine + string.Join("\r\n", result.ToArray());
+            var operand = instruction.Operand[instruction.OperandPrefixSize..].ToArray();
+            var asStr = Encoding.UTF8.GetString(operand);
+            var readable = asStr.All(char.IsAsciiLetterOrDigit);
+            return instruction.OpCode switch
```

Comment:
Move to .ToString()?

Reply:
It does return a `string`. However I am checking it before that.

Reply:
Moved